setup.py

```diff
@@ -49,7 +47,7 @@ def read(*rnames):
     namespace_packages=["plone"],
     include_package_data=True,
     zip_safe=False,
-    python_requires=">=2.7",
+    python_requires=">=3.6",
```

Comment:
Dont we want minimum 3.8 here? It is your call though.

Reply:
Ah, you gave the reason in commit 1740f6f05ec0a5f466716ac35b6648f950045ca1 already.  That is fine.

I have done similar changes for some core 5.2 packages, or maybe add-ons.

Reply:
@mauritsvanrees I was wondering about this as well. Do we want to fail hard or allow folks to use Python 3.6/3.7 but then they are on their own. What's our policy for other core packages?

Reply:
I am ok with it either way. I'd just like us to handle this in a consistent manner.

Reply:
@mauritsvanrees I checked CMFPlone (https://github.com/plone/Products.CMFPlone/blob/master/setup.py#L29) and I will go with python_requires>=3.8 for plone.rest and plone.restapi to be consistent what we do in other packages.

Reply:
Yes, if it is only used in Plone 6.0 and higher, then I would require 3.8 minimum.

And after October next year when Python 3.8 is EOL, we could remove the classifiers, but we would keep `python_requires>=3.8` for branches that still support Plone 6.0.

Although: if we add `check-python-versions` as a linter, then this may complain if there is a mismatch between the classifiers and `python_requires`.  So we could keep the classifiers then.

Python classifiers feels like something we promise to support.
`python_requires` feels like something we promise not to explicitly break.

Reply:
@mauritsvanrees what's the latest Python version that Plone 5.2 still supports? Since we want to continue to support both Plone 5.2 and Plone 6.x at the same time we need at least some overlap. 

I guess we are safe since both Plone 5.2 and Plone 6.x support Python 3.8 (at least we have a Jenkins job):

https://jenkins.plone.org/job/plone-5.2-python-3.8/

The interesting question will be when Python 3.8 reaches its EOL. I guess Plone 5.2 is not tested with Python 3.9, right?

My thinking was that we can drop Python 3.6 and Python 3.7 support and continue to support Plone 5.2. If you run Plone 5.2 with outdated Python versions, you are on your own.

Are you saying that if we want to support both 5.2 and 6.x with plone.rest we should rather keep the minimum requirement to Python 3.6?